ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -768,9 +792,9 @@ boolean getEntriesSince(StoreKey key, FindEntriesCondition findEntriesCondition,
       }
       for (Map.Entry<StoreKey, IndexValue> entry : tempMap.entrySet()) {
         if (key == null || entry.getKey().compareTo(key) != 0) {
-          MessageInfo info = new MessageInfo(entry.getKey(), entry.getValue().getSize(),
-              entry.getValue().isFlagSet(IndexValue.Flags.Delete_Index), entry.getValue().getExpiresAtMs());
-          entries.add(info);
+          IndexValue newValue = new IndexValue(startOffset.getName(), entry.getValue().getBytes(), getVersion());
```

Comment:
I thought it might be safer to create a copy instead of returning the original index value (in case caller modifies it intentionally or unintentionally). Since the index values in this code path haven't got a chance to be persisted yet.